src/test/scala/chiselTests/OneHotMuxSpec.scala

```diff
@@ -0,0 +1,345 @@
+// See LICENSE for license details.
+
+package chiselTests
+
+import Chisel.testers.BasicTester
+import chisel3._
+import chisel3.experimental.FixedPoint
+import chisel3.util.Mux1H
+import org.scalatest._
+
+//scalastyle:off magic.number
+
+class OneHotMuxSpec extends FreeSpec with Matchers with ChiselRunners {
+  "simple one hot mux with uint should work" in {
+    assertTesterPasses(new SimpleOneHotTester)
+  }
+  "simple one hot mux with sint should work" in {
+    assertTesterPasses(new SIntOneHotTester)
+  }
+  "simple one hot mux with fixed point should work" in {
+    assertTesterPasses(new FixedPointOneHotTester)
+  }
+  "simple one hot mux with all same fixed point should work" in {
+    assertTesterPasses(new AllSameFixedPointOneHotTester)
+  }
+  "simple one hot mux with all same parameterized sint values should work" in {
+    val values: Seq[SInt] = Seq((-3).S, (-5).S, (-7).S, (-11).S)
+    assertTesterPasses(new ParameterizedOneHotTester(values, SInt(8.W), -5.S(8.W)))
+  }
+  "simple one hot mux with all same parameterized aggregates containing fixed values should work" in {
+    assertTesterPasses(new ParameterizedAggregateOneHotTester)
+  }
+  "simple one hot mux with all aggregates containing inferred width fixed values should NOT work" in {
+    intercept[ChiselException] {
+      assertTesterPasses(new InferredWidthAggregateOneHotTester)
+    }
+  }
+  "simple one hot mux with all fixed width bundles but with different bundles should Not work" in {
+    intercept[IllegalArgumentException] {
+      assertTesterPasses(new DifferentBundleOneHotTester)
+    }
+  }
+}
+
+class SimpleOneHotTester extends BasicTester {
+  val dut = Module(new SimpleOneHot)
+  dut.io.selectors(0) := false.B
+  dut.io.selectors(1) := false.B
+  dut.io.selectors(2) := true.B
+  dut.io.selectors(3) := false.B
+
+  assert(dut.io.out === 8.U)
+
+  stop()
+}
+
+class SimpleOneHot extends Module {
+  val io = IO(new Bundle {
+    val selectors = Input(Vec(4, Bool()))
+    val out = Output(UInt(8.W))
+  })
+
+  io.out := Mux1H(Seq(
+    io.selectors(0) -> 2.U,
+    io.selectors(1) -> 4.U,
+    io.selectors(2) -> 8.U,
+    io.selectors(3) -> 11.U
+  ))
+}
+class SIntOneHotTester extends BasicTester {
+  val dut = Module(new SIntOneHot)
+  dut.io.selectors(0) := false.B
+  dut.io.selectors(1) := true.B
+  dut.io.selectors(2) := false.B
+  dut.io.selectors(3) := false.B
+
+  printf("out is %d\n", dut.io.out)
```

Comment:
please no printfs in test, no one will actually look at the console output, and most likely it will not make sense with the other 100+ tests in parallel